source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "envoy/common/time.h"
+#include "envoy/http/header_map.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+using OptionalDuration = absl::optional<SystemTime::duration>;
+
+struct RequestCacheControl {
+  RequestCacheControl() = default;
+  explicit RequestCacheControl(absl::string_view cache_control_header);
+
+  // must_validate is true if 'no-cache' directive is present
+  bool must_validate_ : 1;
+  bool no_store_ : 1;
+  // 'no-transform' directive is not used now
+  bool no_transform_ : 1;
+  // 'only-if-cached' directive is not used now
+  bool only_if_cached_ : 1;
+  OptionalDuration max_age_;
+  OptionalDuration min_fresh_;
+  OptionalDuration max_stale_;
+};
+
+struct ResponseCacheControl {
+  ResponseCacheControl() = default;
+  explicit ResponseCacheControl(absl::string_view cache_control_header);
+
+  // must_validate is true if 'no-cache' directive is present; arguments are ignored for now
+  bool must_validate_ : 1;
```

Comment:
is there a compelling reason to bit-pack these? You have 5 bools here which would easily fit into a uint64_t, which I think you'll pay for anyway due to alignment.

Reply:
It was a suggestion of @toddmgreer 's review.

Reply:
Todd what's the advantage of bit-packing these? At first glance it's (a) not smaller due to alignment and (b)  this is a temp structure and it probably doesn't matter how small it is.

Reply:
The `ResponseCacheControl` struct is indeed only used temporarily in one function for now. However, the `RequestCacheControl` struct is a member of `LookupRequest` that lives for a bit longer (for the duration of the cache lookup).

Reply:
Makes sense. Still I want to know what bit-packing buys in terms of overall sizeof this struct. It almost certainly costs you instructions, as well as the ability to nit in the declaration.

Reply:
Apparently, I either can't count, or hallucinated additional member variables that pushed them past 56 and 24 bytes. I rechecked, and indeed the bit-packing has no benefits:
RequestCacheControl is 56 bytes either way, and ResponseCacheControl is 24 bytes either way. Ditch it.